docs/Windows.md

```diff
@@ -1,43 +1,172 @@
-2. You need to add the folder containing `swiftc.exe` to the `PATH` environment variable to use the toolchain.
-3. You also need to add the folder containing the icu dlls to the `PATH` environment variable. 
+SET PATH=%SCRIPT_DIR%\cmake\bin;%PATH%
+
+in the batch script. For the path to CMake, adding it permanently to your PATH environment variable does not really help (try `cmake -version` inside the Native Tools Command Prompt), you would have to set it manually in each instance the Native Tools Command Prompt first, so a batch script might be a good idea.
+
+The following commands build the project mentioned above (including tests), to be executed inside the project directory after the PATH environment variable is set):
 
-This can be achieved by entering the following into the command line.
 ```cmd
-path %PATH%;C:\Library\Developer\Toolchains\unknown-Asserts-development.xctoolchain\usr\bin
-path %PATH%;C:\Library\icu-64.2\bin
+cmake -G Ninja -B build -DCMAKE_BUILD_TYPE=RelWithDebInfo -DBUILD_TESTING=YES
+cd build
+ninja
+ninja test
 ```
 
-Alternatively, to make these changes persistent go to the Windows environment variables edit dialog found in `System Properties` > `Advanced` > `Environment Variables` and edit the path variable to add the following two new items.
+Of course, those commands have to be executed inside the Native Tools Command Prompt.
+
+In a Windows batch script, you might want to add  `|| goto :end` at the end of each of those commands plus an `:end` mark to ensure that this list of commands is interrupted as soon as an error occurs. A full batch script to build the project in `C:\tmp\swift-cmake-demo` then might look as follows (of course, that batch script has to be called from within the Native Tools Command Prompt):
+
+```cmd
+@ECHO OFF
+
+SET HOME=%CD%
+
+SET INSTALLATION_DIR=C:\Library
+
+SET PROJECT_DIR=C:\tmp\swift-cmake-demo
+
+SET PATH=%INSTALLATION_DIR%\Developer\Toolchains\unknown-Asserts-development.xctoolchain\usr\bin;%PATH%
+SET PATH=%INSTALLATION_DIR%\cmake\bin;%PATH%
+SET PATH=%INSTALLATION_DIR%\icu-64.2\bin;%PATH%
+SET PATH=%INSTALLATION_DIR%\icu-63.1\bin;%PATH%
+
+CD "%PROJECT_DIR%"
+
+cmake -G Ninja -B build -DCMAKE_BUILD_TYPE=RelWithDebInfo -DBUILD_TESTING=YES || goto :end
+cd build || goto :end
+ninja || goto :end
+ninja test || goto :end
+
+:end
+CD "%HOME%"
+```
+
+#### Running the Swift program on the development machine
+
+The files inside  `C:\Library\Swift\Current\bin` are needed to run the program on the  development machine, so `C:\Library\Swift\Current\bin` has to be added to the PATH environment variable in order to run the program.
+
+Additionally, the DLL files from the ICU project (icudt*.dll, icuin*.dll, icuio*.dll, icutu*.dll, icuuc*.dll) have to be in your path.
+
+So a batch script to run the project in `C:\tmp\swift-cmake-demo` may look as follows:
```

Comment:
Again, I don't think that we should be encouraging batch files.  We should recommend that users modify the `PATH` variable or better yet, we actually package the needed libraries into the image.